File: OEE-API/Models/SYF/ActionTime.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace OEE_API.Models.SYF
{
    public class ActionTime
    {
        public int ID { get; set; }

        [StringLength(50)]
        public string Factory_ID { get; set; }

        [StringLength(50)]
        public string Machine_ID { get; set; }

        public DateTime Shiftdate { get; set; }
        public DateTime Start_Time { get; set; }
        public DateTime End_Time { get; set; }
        public int Duration { get; set; }
        public int Shift { get; set; }
        public int Status_ID { get; set; }
        [StringLength(10)]
        public string Status { get; set; }

    }
}

File: OEE-SPA/src/app/_core/_models/action-time.ts
export interface ActionTime {
    id: number;
    factory_id: string;
    machine_id: string;
    building_id: string;
    shiftdate?: Date;
    date: string;
    start_time?: Date;
    end_time?: Date;
    duration: string;
    status_id: string;
    status: string;
    shift: string;
}
